### PR TITLE
FEM-2105 IMADAI plugin support

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java [hunk 1]
```diff
@@ -0,0 +1,1222 @@
...
+                break;
+            case AD_BREAK_ENDED: //Fired when an ad break ends.
+                log.d("AD AD_BREAK_ENDED");
+                messageBus.post(new AdEvent(AdEvent.Type.AD_BREAK_ENDED));
+                messageBus.post(new AdEvent(AdEvent.Type.CONTENT_RESUME_REQUESTED));
+                boolean allAdsPlayed = true;
+                boolean lastAdPlayed = true;
+                List<CuePoint> cuesList = streamManager.getCuePoints();
+                if (cuesList != null) {
+                    for (int i = 0 ; i < cuesList.size() ; i++) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: Use Iterator to iterate here.

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java [hunk 2]
```diff
@@ -0,0 +1,1222 @@
...
+                        if (!cuesList.get(i).isPlayed()) {
+                            allAdsPlayed = false;
+                        }
+                        if (i == cuesList.size() -1) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: Use iterator.hasNext() to detect the last item in the list.

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java [hunk 3]
```diff
@@ -0,0 +1,1222 @@
...
+        com.kaltura.playkit.plugins.ads.AdEvent errorEvent = new com.kaltura.playkit.plugins.ads.AdEvent.Error(new PKError(errorType, message, exception));
+        messageBus.post(errorEvent);
+    }
+
+    private void sendCuePointsUpdate() {
+        log.d("sendCuePointsUpdate");
+        List<Long> cuePointsList = buildCuePointsList();
+        if (cuePointsList == null)
+            return;
+        if (cuePointsList.size() > 0) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: use List.isEmpty()

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java [hunk 4]
```diff
@@ -0,0 +1,1222 @@
...
+                            }
+                        }
+                        messageBus.removeListener(this);
+                    }
+                });
+            }
+        }
+    }
+
+//    private PKMediaConfig createMediaConfig(String url) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: Remove commented-out code.

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java (hunk 1 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: done

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java (hunk 2 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: done

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java (hunk 3 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: done for all other same conditioons

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java (hunk 4 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java]: done

**Review by noamtamim — APPROVED** [review]

**Review by noamtamim — APPROVED** [review]

**Pull request merged**